### DISCUSSION TITLE
Is this the proper use of the gui.broadcast method?

### DISCUSSION THREAD

**Discussion by rseward** [Avaiga/taipy#287] [Q&A]
I am trying to write a simple server to show the uptime of the system hosting the taipy gui server. Is this the proper use of the broadcast method?

If it is, it is NOT giving the expected result. I expected the uptime to update on all connected clients. Failing that, the uptime var should at least change on the client page reload. 

The project looks very promising. I also applaud the name! Please continue the great work!

```
#!/usr/bin/env python

from taipy import Gui

import subprocess
import time

def mysys(args):
    myargs=args
    if type(args) == str:
        myargs=[ args ]
    
    proc = subprocess.run( myargs, shell=True, stdout=subprocess.PIPE, stderr=subprocess.STDOUT, encoding='UTF-8' )

    return proc.stdout

uptime = mysys("uptime")

main="""Hello World!

## Uptime

<|{uptime}|>
"""

mygui = Gui(page=main)
mygui.run(dark_mode=True, run_in_thread=True)

while True:
    print(uptime)
    uptime = mysys( "uptime" )
    mygui.broadcast( "uptime", uptime )
    #mygui.reload() # opens a new browser tab!
    time.sleep(1)
```

**Comment by FlorianJacta** [comment]
Broadcasting with 2.4 is quite complex, I would suggest waiting for the next release 3.0. You could also use the develop branch to try it now. To download, the develop version of Taipy Gui:
- Install git on your environment (`conda install git` for conda environment),
- [node.js](https://nodejs.org/en) on your computer
- `pip install git+https://github.com/Avaiga/taipy-gui.git`

You define this variable to be shared by everyone and it will be automatically updated to every client.

```python
from taipy.gui import Gui, State

def on_action(state: State):
    state.nb = state.nb+1

nb = 0

md = """
<|{nb}|> <|Refresh|button|on_action=on_action|>
"""

gui = Gui(page=md)
gui.add_shared_variable("nb")
gui.run()
```
